### PR TITLE
Remove transform

### PR DESCRIPTION
**Pull request by Elad-Laufer**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -4,7 +4,7 @@ root = true
 charset = utf-8
 end_of_line = lf
 indent_style = tab
-indent_size = 2
+indent_size = 4
```
Comment (golanpa) [inline .editorconfig]: thank god

File: vips/color.c [hunk 2]
```diff
@@ -7,3 +9,31 @@ int is_colorspace_supported(VipsImage *in) {
 int to_colorspace(VipsImage *in, VipsImage **out, VipsInterpretation space) {
 	return vips_colourspace(in, out, space, NULL);
 }
+
+// https://libvips.github.io/libvips/API/8.6/libvips-colour.html#vips-icc-transform
+int optimize_icc_profile(VipsImage *in, VipsImage **out, int isCmyk) {
+	// todo: check current embedded profile, and skip if already set
+
+    char *srgb_profile_path = SRGB_V2_MICRO_ICC_PATH;  // SRGB_IEC61966_2_1_ICC_PATH
```
Comment (golanpa) [inline vips/color.c]:
```suggestion
    char *srgb_profile_path = SRGB_V2_MICRO_ICC_PATH;
```

File: vips/color.c [hunk 3]
```diff
@@ -7,3 +9,31 @@ int is_colorspace_supported(VipsImage *in) {
...
+    char *gray_profile_path = SGRAY_V2_MICRO_ICC_PATH;  // GENERIC_GRAY_GAMMA_2_2_ICC_PATH
```
Comment (golanpa) [inline vips/color.c]:
```suggestion
    char *gray_profile_path = SGRAY_V2_MICRO_ICC_PATH;
```

Comment (golanpa) [reply vips/color.c]: removed the remark

File: vips/color.c (hunk 2 above)
Comment (golanpa) [reply vips/color.c]: removed the remark

File: vips/color.c [hunk 4]
```diff
@@ -7,3 +9,31 @@ int is_colorspace_supported(VipsImage *in) {
 int to_colorspace(VipsImage *in, VipsImage **out, VipsInterpretation space) {
 	return vips_colourspace(in, out, space, NULL);
 }
+
+// https://libvips.github.io/libvips/API/8.6/libvips-colour.html#vips-icc-transform
+int optimize_icc_profile(VipsImage *in, VipsImage **out, int isCmyk) {
```
Comment (golanpa) [inline vips/color.c]: didn't you want to remove the _isCmyk_ parameter?

File: vips/color.c [hunk 5]
```diff
@@ -7,3 +9,31 @@ int is_colorspace_supported(VipsImage *in) {
...
+    int channels = vips_image_get_bands(in);
+	int result;
+
+    if (channels > 2) {
+    	if (isCmyk == 1) {
+    		result = vips_icc_transform(in, out, srgb_profile_path, "input_profile", "cmyk", "intent", VIPS_INTENT_PERCEPTUAL, NULL);
+    	} else {
+        	result = vips_icc_transform(in, out, srgb_profile_path, "embedded", TRUE, "intent", VIPS_INTENT_PERCEPTUAL, NULL);
+        	// ignore embedded errors
+        	if (result != 0) {
```
Comment (golanpa) [inline vips/color.c]: why to ignore embedded errors? should we ignore other as well?

File: vips/foreign.c [hunk 6]
```diff
@@ -1,5 +1,6 @@
 #include "lang.h"
 #include "foreign.h"
+#include <stdio.h>
```
Comment (golanpa) [inline vips/foreign.c]: what's the reason? eliminate compile warnings?

File: vips/header.c [hunk 7]
```diff
@@ -39,7 +14,7 @@ void remove_metadata(VipsImage *in) {
     gchar ** fields = vips_image_get_fields(in);
 
     for (int i=0; fields[i] != NULL; i++) {
-        if (strncmp(fields[i], VIPS_META_ICC_NAME, 16)) {
+        if (strncmp(fields[i], VIPS_META_ICC_NAME, 16) && strncmp(fields[i], VIPS_META_ORIENTATION, 16)) {
```
Comment (golanpa) [inline vips/header.c]: from some reason I cannot find the definition for VIPS_META_ORIENTATION - please make sure that 16 is the size you want to compare and that 16 is not high.

File: vips/image_golden_test.go [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+	})
+}
+
+func TestImage_Resize_Upscale_Alpha(t *testing.T) {
+	goldenTest(t, resources+"png-8bit+alpha.png", nil, func(img *ImageRef) error {
+		return img.Resize(1.1, KernelLanczos3)
+	})
+}
+
+func TestImage_OptimizeICCProfile_CMYK(t *testing.T) {
```
Comment (golanpa) [inline vips/image_golden_test.go]: need to check that our profile was really set

File: vips/image_golden_test.go [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+func TestImage_Resize_Downscale_Alpha(t *testing.T) {
+	goldenTest(t, resources+"png-8bit+alpha.png", nil, func(img *ImageRef) error {
+		return img.Resize(0.9, KernelLanczos3)
+	})
+}
+
+func TestImage_Resize_Upscale_Alpha(t *testing.T) {
+	goldenTest(t, resources+"png-8bit+alpha.png", nil, func(img *ImageRef) error {
+		return img.Resize(1.1, KernelLanczos3)
```
Comment (golanpa) [inline vips/image_golden_test.go]: maybe check the dimensions of the resultant image

Comment (golanpa) [reply vips/image_golden_test.go]: and for other tests as well

File: vips/image_golden_test.go [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+	})
+}
+
+func TestImage_RemoveICCProfile(t *testing.T) {
+	goldenTest(t, resources+"jpg-24bit-icc-smpte.jpg", nil, func(img *ImageRef) error {
+		return img.RemoveICCProfile()
+	})
+}
+
+func TestImage_RemoveMetadata(t *testing.T) {
```
Comment (golanpa) [inline vips/image_golden_test.go]: maybe check that metadata was removed

File: vips/color.c (hunk 4 above)
Comment (Elad-Laufer) [reply vips/color.c]: eventually, didn't want to mess to much with the C code at once

File: vips/color.c (hunk 5 above)
Comment (Elad-Laufer) [reply vips/color.c]: The others are legit errors, as a profile is provided from outside.

File: vips/foreign.c (hunk 6 above)
Comment (Elad-Laufer) [reply vips/foreign.c]: I don't know... Not a C master

File: vips/header.c (hunk 7 above)
Comment (Elad-Laufer) [reply vips/header.c]: ` #define VIPS_META_ORIENTATION "orientation"`

It's just for safety

File: vips/image_golden_test.go (hunk 8 above)
Comment (Elad-Laufer) [reply vips/image_golden_test.go]: Take a look at the golden image, either it's there and we're good, or it is not and we have a bug

File: vips/image_golden_test.go (hunk 9 above)
Comment (Elad-Laufer) [reply vips/image_golden_test.go]: Golden test, need to verify once the golden image

File: vips/image_golden_test.go (hunk 10 above)
Comment (Elad-Laufer) [reply vips/image_golden_test.go]: Again, golden test, check once on the golden image

File: vips/header.c (hunk 7 above)
Comment (golanpa) [reply vips/header.c]: try using:
sizeof(VIPS_META_ICC_NAME), sizeof(VIPS_META_ORIENTATION) instead the magic number 16

File: vips/image_golden_test.go (hunk 8 above)
Comment (golanpa) [reply vips/image_golden_test.go]: but golden are checked visually, no one pay attention to the profile

File: vips/foreign.c (hunk 6 above)
Comment (golanpa) [reply vips/foreign.c]: so let's remove it if it is not neccessary

File: vips/image.go [hunk 11]
```diff
@@ -414,9 +424,25 @@ func (r *ImageRef) RemoveICCProfile() error {
...
-	out, err := vipsTransformICCProfile(r.image, isCmyk)
+	out, err := vipsOptimizeICCProfile(r.image, isCmyk)
+	if err != nil {
+		return err
+	}
+	r.setImage(out)
+	return nil
+}
+
+func (r *ImageRef) OptimizeICCProfile() error {
```
Comment (golanpa) [inline vips/image.go]: why do we need both OptimizeICCProfile and TransformICCProfile ?

Comment (Elad-Laufer) [reply vips/image.go]: backwards compatibility until I kill the old function

File: vips/foreign.c (hunk 6 above)
Comment (Elad-Laufer) [reply vips/foreign.c]: done

File: vips/header.c (hunk 7 above)
Comment (Elad-Laufer) [reply vips/header.c]: done

**Pull request merged**